native_extension/GoogleSignInANE/src/com/tuarua/google/signin/GoogleSignInOptions.as

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2021 Tua Rua Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tuarua.google.signin {
+
+/**
+ * Options for GoogleSignIn client.
+ */
+public class GoogleSignInOptions {
+
+    /**
+     * Default configuration for Google Sign In.
+     */
+    public static const DEFAULT_SIGN_IN:String = 'DEFAULT_SIGN_IN';
+
+    /**
+     * Default and recommended configuration for Games Sign In.
+     */
+    public static const DEFAULT_GAMES_SIGN_IN:String = 'DEFAULT_GAMES_SIGN_IN';
+
+    private var _base:String;
+    private var _requestEmail:Boolean;
+    private var _requestId:Boolean;
+    private var _requestProfile:Boolean;
+    private var _requestIdToken:Boolean;
+    private var _requestServerAuthCode:Boolean;
+    private var _serverClientId:String;
+    private var _requestScopes:Array;
+
+    /**
+     * You can specify just needed properties.
+     *
+     * <listing version="3.0">
+     * var options:GoogleSignInOptions = new GoogleSignInOptions({
+     *     base: GoogleSignInOptions.DEFAULT_GAMES_SIGN_IN,
+     *     requestServerAuthCode: true
+     * });
+     * </listing>
+     *
+     * @param {Object} [options = {}]
+     * @param {String} [options.base = 'DEFAULT_SIGN_IN']
+     * @param {Boolean} [options.requestEmail = false]
+     * @param {Boolean} [options.requestId = false]
+     * @param {Boolean} [options.requestProfile = false]
+     * @param {Boolean} [options.requestIdToken = false]
+     * @param {Boolean} [options.requestServerAuthCode = false]
+     * @param {String} [options.serverClientId = null]
+     * @param {String[]} [options.requestScopes = []]
+     */
+    public function GoogleSignInOptions(options:Object = null) {
```

Comment:
Same as mentioned above
I don't like passing objects into constructors.
Getters and setters should be used with default property values set when the private var is declared.

See 
https://github.com/tuarua/Firebase-ANE/blob/master/native_extension/FirestoreANE/src/com/tuarua/firebase/firestore/FirestoreSettings.as

Reply:
Google Sign In has fewer options on the Swift side, and Play Games is not supported officially.